### ISSUE TITLE
Database SDK tries to access credential even when FIREBASE_DATABASE_EMULATOR_HOST is set

### ISSUE THREAD

**Issue by filipesilva** [firebase/firebase-js-sdk#3144]
### [REQUIRED] Environment info

**firebase-tools:** 8.4.0

**Platform:** Windows

### [REQUIRED] Test case

I have a test case with minimal cloud function code and emulator at https://github.com/filipesilva/firebase-emulator-credentials.
```
const functions = require('firebase-functions');
const admin = require('firebase-admin');
admin.initializeApp();

exports.helloWorld = functions.https.onRequest(async (request, response) => {
  const db = admin.database();
  const ref = await db.ref('auth-test').set({
    date: new Date().toISOString()
  });
  response.send(`Added: ${ref.id}`);
});
```

### [REQUIRED] Steps to reproduce

```
git clone https://github.com/filipesilva/firebase-emulator-credentials
cd firebase-emulator-credentials
yarn
yarn firebase logout
yarn emulators
# open http://localhost:5001/emulator-test-1/us-central1/helloWorld in the browser
```

You should see the following error log in the console:
```
kamik@RED-X1C6 MINGW64 /d/sandbox/firebase-emulator-credentials
$ yarn emulators
yarn run v1.21.1
$ firebase emulators:start
!  emulators: You are not currently authenticated so some features may not work correctly. Please run firebase login to authenticate the CLI.
i  emulators: Starting emulators: functions, database
!  Your requested "node" version "8" doesn't match your global version "12"
i  database: Database Emulator logging to database-debug.log
!  hosting: The hosting emulator is configured but there is no hosting configuration. Have you run firebase init hosting?
i  ui: Emulator UI logging to ui-debug.log
i  functions: Watching "D:\sandbox\firebase-emulator-credentials\functions" for Cloud Functions...
+  functions[helloWorld]: http function initialized (http://localhost:5001/emulator-test-1/us-central1/helloWorld).

┌───────────────────────────────────────────────────────────────────────┐
│ ✔  All emulators ready! View status and logs at http://localhost:4000 │
└───────────────────────────────────────────────────────────────────────┘

┌───────────┬────────────────┬─────────────────────────────────┐
│ Emulator  │ Host:Port      │ View in Emulator UI             │
├───────────┼────────────────┼─────────────────────────────────┤
│ Functions │ localhost:5001 │ http://localhost:4000/functions │
├───────────┼────────────────┼─────────────────────────────────┤
│ Database  │ localhost:9000 │ http://localhost:4000/database  │
└───────────┴────────────────┴─────────────────────────────────┘
  Other reserved ports: 4400, 4500

Issues? Report them at https://github.com/firebase/firebase-tools/issues and attach the *-debug.log files.

i  functions: Beginning execution of "helloWorld"
>  [2020-05-26T15:20:21.433Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:22.307Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:23.270Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:24.471Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:26.454Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:28.169Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:30.516Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:33.439Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:38.337Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:39.686Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:48.691Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:20:51.557Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
>  [2020-05-26T15:21:05.098Z]  @firebase/database: FIREBASE WARNING: {"code":"app/invalid-credential","message":"Credential implementation provided to initializeApp() via the
 \"credential\" property failed to fetch a valid Google OAuth2 access token with the following error: \"Error fetching access token: Error while making request: getaddrinfo E
NOTFOUND metadata.google.internal. Error code: ENOTFOUND\"."}
!  functions: Your function timed out after ~60s. To configure this timeout, see
      https://firebase.google.com/docs/functions/manage-functions#set_timeout_and_memory_allocation.
>  D:\sandbox\firebase-emulator-credentials\node_modules\firebase-tools\lib\emulator\functionsEmulatorRuntime.js:619
>                  throw new Error("Function timed out.");
>                  ^
>
>  Error: Function timed out.
>      at Timeout._onTimeout (D:\sandbox\firebase-emulator-credentials\node_modules\firebase-tools\lib\emulator\functionsEmulatorRuntime.js:619:23)
>      at listOnTimeout (internal/timers.js:531:17)
>      at processTimers (internal/timers.js:475:7)
```

### [REQUIRED] Expected behavior

The emulated cloud function call should be successful even though there are no credentials.

### [REQUIRED] Actual behavior

The emulated cloud function errors out.

### Extra details:

This is a followup from @samtstern's comment on https://github.com/firebase/firebase-tools/issues/1940#issuecomment-634062155.

It's currently not possible to use `firebase-admin` with the emulators without credentials obtained via https://cloud.google.com/docs/authentication/getting-started.

This use case is especially relevant for CI, where there is no user to login and no credentials should be provided, especially on CI test runners for open-source projects that may receive PRs from forks and which to test them.

**Comment by avolkovi** [comment]
reassigning to @schmidt-sebastian since this will require changing the RTDB SDK

**Comment by filipesilva** [comment]
@schmidt-sebastian heya, apologies for bugging you about this, but have you had time to look at it? It's still blocking us from updating the emulators since then the CI will break.

**Comment by schmidt-sebastian** [comment]
I haven't yet. I will try to get to this this week.

**Comment by filipesilva** [comment]
Thanks for taking the time @schmidt-sebastian. If there's anything I can do to help, or things you'd like me to test, let me know.

**Comment by filipesilva** [comment]
Hi all, just wanted to mention that I've just tried updating and I can confirm it unblocks us, cheers!

It's still possible to get an old version of RTDB while installing `firebase-admin@9.0.0` since it's dependency is set as `"@firebase/database": "^0.6.0"`, but I think this fix was only released on `"@firebase/database@0.6.8"`. 

Clearing `yarn.lock`/`package.lock` can trigger an update, as can using the `resolutions` field in package.json if using yarn.

Thanks for all the work in getting this in!

**Comment by samtstern** [comment]
@filipesilva thank you for confirming that this works!
